File: beginner/reverse-array-in-place/reverse-array-in-place.py
import math

def reverse_array_in_place(arr):
  half_len = int(math.ceil(len(arr)/2))
  for index in range(half_len):
    swap_index = len(arr) - 1 - index
    # don't need temp vars in python, can do multiple assignment
    arr[index], arr[swap_index] = arr[swap_index], arr[index]

  return arr

print(reverse_array_in_place([1, 3, 5, 7, 9]))
print(reverse_array_in_place([1, 3, 5, 7, 9, 11]))

File: easy/reverse-array-in-place/reverse-array-in-place.ts
function reverseArrayInPlace(arr: any[]): any[] {
  for (let i = 0; i < arr.length / 2; i++) {
    const temp = arr[i];
    const swapIndex = arr.length - 1 - i;
    arr[i] = arr[swapIndex];
    arr[swapIndex] = temp;
  }
  return arr;
}

console.log(reverseArrayInPlace([1, 3, 5, 7, 9]));
console.log(reverseArrayInPlace([1, 3, 5, 7, 9, 11]));
